<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Package Functions Help – EnrichGT</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-1f64a611174d0987db8176e27c99ed25.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./EnrichGTlogo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">EnrichGT</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./pkgfuns.html" aria-current="page"> 
<span class="menu-text">Package Functions</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/ZhimingYe/EnrichGT/"> 
<span class="menu-text">Github</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./preface.html"> 
<span class="menu-text">Motivation</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./acknowledgement.html"> 
<span class="menu-text">Acknowledgement</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#description" id="toc-description" class="nav-link active" data-scroll-target="#description">DESCRIPTION</a></li>
  <li><a href="#convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys" id="toc-convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys" class="nav-link" data-scroll-target="#convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys"><code>convert_annotations_genes</code>: Convert gene annotations from any keys to any keys</a>
  <ul class="collapse">
  <li><a href="#description-1" id="toc-description-1" class="nav-link" data-scroll-target="#description-1">Description</a></li>
  <li><a href="#usage" id="toc-usage" class="nav-link" data-scroll-target="#usage">Usage</a></li>
  <li><a href="#arguments" id="toc-arguments" class="nav-link" data-scroll-target="#arguments">Arguments</a></li>
  <li><a href="#value" id="toc-value" class="nav-link" data-scroll-target="#value">Value</a></li>
  </ul></li>
  <li><a href="#database_from_gmt-parse-gmt-format-gene-set-files" id="toc-database_from_gmt-parse-gmt-format-gene-set-files" class="nav-link" data-scroll-target="#database_from_gmt-parse-gmt-format-gene-set-files"><code>database_from_gmt</code>: Parse GMT format gene set files</a>
  <ul class="collapse">
  <li><a href="#description-2" id="toc-description-2" class="nav-link" data-scroll-target="#description-2">Description</a></li>
  <li><a href="#usage-1" id="toc-usage-1" class="nav-link" data-scroll-target="#usage-1">Usage</a></li>
  <li><a href="#arguments-1" id="toc-arguments-1" class="nav-link" data-scroll-target="#arguments-1">Arguments</a></li>
  <li><a href="#author" id="toc-author" class="nav-link" data-scroll-target="#author">Author</a></li>
  <li><a href="#value-1" id="toc-value-1" class="nav-link" data-scroll-target="#value-1">Value</a></li>
  <li><a href="#examples" id="toc-examples" class="nav-link" data-scroll-target="#examples">Examples</a></li>
  </ul></li>
  <li><a href="#egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing" id="toc-egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing" class="nav-link" data-scroll-target="#egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing"><code>egt_compare_groups</code>: 2-Group Comparison of enrichment results and further clustering and visualizing</a>
  <ul class="collapse">
  <li><a href="#description-3" id="toc-description-3" class="nav-link" data-scroll-target="#description-3">Description</a></li>
  <li><a href="#usage-2" id="toc-usage-2" class="nav-link" data-scroll-target="#usage-2">Usage</a></li>
  <li><a href="#arguments-2" id="toc-arguments-2" class="nav-link" data-scroll-target="#arguments-2">Arguments</a></li>
  <li><a href="#details" id="toc-details" class="nav-link" data-scroll-target="#details">Details</a></li>
  <li><a href="#author-1" id="toc-author-1" class="nav-link" data-scroll-target="#author-1">Author</a></li>
  <li><a href="#value-2" id="toc-value-2" class="nav-link" data-scroll-target="#value-2">Value</a></li>
  </ul></li>
  <li><a href="#egt_enrichment_analysis-perform-over-representation-analysis-ora" id="toc-egt_enrichment_analysis-perform-over-representation-analysis-ora" class="nav-link" data-scroll-target="#egt_enrichment_analysis-perform-over-representation-analysis-ora"><code>egt_enrichment_analysis</code>: Perform Over-Representation Analysis (ORA)</a>
  <ul class="collapse">
  <li><a href="#description-4" id="toc-description-4" class="nav-link" data-scroll-target="#description-4">Description</a></li>
  <li><a href="#usage-3" id="toc-usage-3" class="nav-link" data-scroll-target="#usage-3">Usage</a></li>
  <li><a href="#arguments-3" id="toc-arguments-3" class="nav-link" data-scroll-target="#arguments-3">Arguments</a></li>
  <li><a href="#details-1" id="toc-details-1" class="nav-link" data-scroll-target="#details-1">Details</a></li>
  <li><a href="#author-2" id="toc-author-2" class="nav-link" data-scroll-target="#author-2">Author</a></li>
  <li><a href="#value-3" id="toc-value-3" class="nav-link" data-scroll-target="#value-3">Value</a></li>
  <li><a href="#examples-1" id="toc-examples-1" class="nav-link" data-scroll-target="#examples-1">Examples</a></li>
  </ul></li>
  <li><a href="#egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea" id="toc-egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea" class="nav-link" data-scroll-target="#egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea"><code>egt_gsea_analysis</code>: Perform Gene Set Enrichment Analysis (GSEA)</a>
  <ul class="collapse">
  <li><a href="#description-5" id="toc-description-5" class="nav-link" data-scroll-target="#description-5">Description</a></li>
  <li><a href="#usage-4" id="toc-usage-4" class="nav-link" data-scroll-target="#usage-4">Usage</a></li>
  <li><a href="#arguments-4" id="toc-arguments-4" class="nav-link" data-scroll-target="#arguments-4">Arguments</a></li>
  <li><a href="#details-2" id="toc-details-2" class="nav-link" data-scroll-target="#details-2">Details</a></li>
  <li><a href="#author-3" id="toc-author-3" class="nav-link" data-scroll-target="#author-3">Author</a></li>
  <li><a href="#value-4" id="toc-value-4" class="nav-link" data-scroll-target="#value-4">Value</a></li>
  <li><a href="#examples-2" id="toc-examples-2" class="nav-link" data-scroll-target="#examples-2">Examples</a></li>
  </ul></li>
  <li><a href="#egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules" id="toc-egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules" class="nav-link" data-scroll-target="#egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules"><code>egt_infer_act</code>: Infering Pathway or Transcript Factors activity from EnrichGT meta-gene modules</a>
  <ul class="collapse">
  <li><a href="#description-6" id="toc-description-6" class="nav-link" data-scroll-target="#description-6">Description</a></li>
  <li><a href="#usage-5" id="toc-usage-5" class="nav-link" data-scroll-target="#usage-5">Usage</a></li>
  <li><a href="#arguments-5" id="toc-arguments-5" class="nav-link" data-scroll-target="#arguments-5">Arguments</a></li>
  <li><a href="#author-4" id="toc-author-4" class="nav-link" data-scroll-target="#author-4">Author</a></li>
  <li><a href="#value-5" id="toc-value-5" class="nav-link" data-scroll-target="#value-5">Value</a></li>
  </ul></li>
  <li><a href="#egt_llm_summary-summarize-enrichgt-results-using-llm" id="toc-egt_llm_summary-summarize-enrichgt-results-using-llm" class="nav-link" data-scroll-target="#egt_llm_summary-summarize-enrichgt-results-using-llm"><code>egt_llm_summary</code>: Summarize EnrichGT results using LLM</a>
  <ul class="collapse">
  <li><a href="#description-7" id="toc-description-7" class="nav-link" data-scroll-target="#description-7">Description</a></li>
  <li><a href="#usage-6" id="toc-usage-6" class="nav-link" data-scroll-target="#usage-6">Usage</a></li>
  <li><a href="#arguments-6" id="toc-arguments-6" class="nav-link" data-scroll-target="#arguments-6">Arguments</a></li>
  <li><a href="#note" id="toc-note" class="nav-link" data-scroll-target="#note">Note</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  <li><a href="#seealso" id="toc-seealso" class="nav-link" data-scroll-target="#seealso">Seealso</a></li>
  <li><a href="#value-6" id="toc-value-6" class="nav-link" data-scroll-target="#value-6">Value</a></li>
  <li><a href="#examples-3" id="toc-examples-3" class="nav-link" data-scroll-target="#examples-3">Examples</a></li>
  </ul></li>
  <li><a href="#egt_plot_gsea-generate-gsea-enrichment-plots" id="toc-egt_plot_gsea-generate-gsea-enrichment-plots" class="nav-link" data-scroll-target="#egt_plot_gsea-generate-gsea-enrichment-plots"><code>egt_plot_gsea</code>: Generate GSEA Enrichment Plots</a>
  <ul class="collapse">
  <li><a href="#description-8" id="toc-description-8" class="nav-link" data-scroll-target="#description-8">Description</a></li>
  <li><a href="#usage-7" id="toc-usage-7" class="nav-link" data-scroll-target="#usage-7">Usage</a></li>
  <li><a href="#arguments-7" id="toc-arguments-7" class="nav-link" data-scroll-target="#arguments-7">Arguments</a></li>
  <li><a href="#author-5" id="toc-author-5" class="nav-link" data-scroll-target="#author-5">Author</a></li>
  <li><a href="#value-7" id="toc-value-7" class="nav-link" data-scroll-target="#value-7">Value</a></li>
  </ul></li>
  <li><a href="#egt_plot_results-visualize-enrichment-results-using-simple-plot" id="toc-egt_plot_results-visualize-enrichment-results-using-simple-plot" class="nav-link" data-scroll-target="#egt_plot_results-visualize-enrichment-results-using-simple-plot"><code>egt_plot_results</code>: Visualize enrichment results using simple plot</a>
  <ul class="collapse">
  <li><a href="#description-9" id="toc-description-9" class="nav-link" data-scroll-target="#description-9">Description</a></li>
  <li><a href="#usage-8" id="toc-usage-8" class="nav-link" data-scroll-target="#usage-8">Usage</a></li>
  <li><a href="#arguments-8" id="toc-arguments-8" class="nav-link" data-scroll-target="#arguments-8">Arguments</a></li>
  <li><a href="#author-6" id="toc-author-6" class="nav-link" data-scroll-target="#author-6">Author</a></li>
  <li><a href="#value-8" id="toc-value-8" class="nav-link" data-scroll-target="#value-8">Value</a></li>
  </ul></li>
  <li><a href="#egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results" id="toc-egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results" class="nav-link" data-scroll-target="#egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results"><code>egt_recluster_analysis</code>: Cluster and re-enrichment enrichment results</a>
  <ul class="collapse">
  <li><a href="#description-10" id="toc-description-10" class="nav-link" data-scroll-target="#description-10">Description</a></li>
  <li><a href="#usage-9" id="toc-usage-9" class="nav-link" data-scroll-target="#usage-9">Usage</a></li>
  <li><a href="#arguments-9" id="toc-arguments-9" class="nav-link" data-scroll-target="#arguments-9">Arguments</a></li>
  <li><a href="#details-3" id="toc-details-3" class="nav-link" data-scroll-target="#details-3">Details</a></li>
  <li><a href="#author-7" id="toc-author-7" class="nav-link" data-scroll-target="#author-7">Author</a></li>
  <li><a href="#value-9" id="toc-value-9" class="nav-link" data-scroll-target="#value-9">Value</a></li>
  <li><a href="#examples-4" id="toc-examples-4" class="nav-link" data-scroll-target="#examples-4">Examples</a></li>
  </ul></li>
  <li><a href="#genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis" id="toc-genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis" class="nav-link" data-scroll-target="#genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis"><code>genes_with_weights</code>: Create a ranked gene list for GSEA analysis</a>
  <ul class="collapse">
  <li><a href="#description-11" id="toc-description-11" class="nav-link" data-scroll-target="#description-11">Description</a></li>
  <li><a href="#usage-10" id="toc-usage-10" class="nav-link" data-scroll-target="#usage-10">Usage</a></li>
  <li><a href="#arguments-10" id="toc-arguments-10" class="nav-link" data-scroll-target="#arguments-10">Arguments</a></li>
  <li><a href="#author-8" id="toc-author-8" class="nav-link" data-scroll-target="#author-8">Author</a></li>
  <li><a href="#value-10" id="toc-value-10" class="nav-link" data-scroll-target="#value-10">Value</a></li>
  <li><a href="#examples-5" id="toc-examples-5" class="nav-link" data-scroll-target="#examples-5">Examples</a></li>
  </ul></li>
  <li><a href="#database_...-get-database-for-enrichment-analysis" id="toc-database_...-get-database-for-enrichment-analysis" class="nav-link" data-scroll-target="#database_...-get-database-for-enrichment-analysis"><code>database_...</code>: Get database for enrichment analysis</a>
  <ul class="collapse">
  <li><a href="#description-12" id="toc-description-12" class="nav-link" data-scroll-target="#description-12">Description</a></li>
  <li><a href="#usage-11" id="toc-usage-11" class="nav-link" data-scroll-target="#usage-11">Usage</a></li>
  <li><a href="#arguments-11" id="toc-arguments-11" class="nav-link" data-scroll-target="#arguments-11">Arguments</a></li>
  <li><a href="#author-9" id="toc-author-9" class="nav-link" data-scroll-target="#author-9">Author</a></li>
  <li><a href="#value-11" id="toc-value-11" class="nav-link" data-scroll-target="#value-11">Value</a></li>
  </ul></li>
  <li><a href="#database_kegg-get-kegg-database-from-kegg-website" id="toc-database_kegg-get-kegg-database-from-kegg-website" class="nav-link" data-scroll-target="#database_kegg-get-kegg-database-from-kegg-website"><code>database_KEGG</code>: Get KEGG database from KEGG website</a>
  <ul class="collapse">
  <li><a href="#description-13" id="toc-description-13" class="nav-link" data-scroll-target="#description-13">Description</a></li>
  <li><a href="#usage-12" id="toc-usage-12" class="nav-link" data-scroll-target="#usage-12">Usage</a></li>
  <li><a href="#arguments-12" id="toc-arguments-12" class="nav-link" data-scroll-target="#arguments-12">Arguments</a></li>
  </ul></li>
  <li><a href="#delete--filter-enrichment-results-by-description-pattern" id="toc-delete--filter-enrichment-results-by-description-pattern" class="nav-link" data-scroll-target="#delete--filter-enrichment-results-by-description-pattern"><code>%-delete-&gt;%</code>: Filter Enrichment Results by Description Pattern</a>
  <ul class="collapse">
  <li><a href="#description-14" id="toc-description-14" class="nav-link" data-scroll-target="#description-14">Description</a></li>
  <li><a href="#usage-13" id="toc-usage-13" class="nav-link" data-scroll-target="#usage-13">Usage</a></li>
  <li><a href="#arguments-13" id="toc-arguments-13" class="nav-link" data-scroll-target="#arguments-13">Arguments</a></li>
  <li><a href="#details-4" id="toc-details-4" class="nav-link" data-scroll-target="#details-4">Details</a></li>
  <li><a href="#value-12" id="toc-value-12" class="nav-link" data-scroll-target="#value-12">Value</a></li>
  <li><a href="#examples-6" id="toc-examples-6" class="nav-link" data-scroll-target="#examples-6">Examples</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/pkgfuns.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Package Functions Help</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="description" class="level1">
<h1>DESCRIPTION</h1>
<pre><code>Type: Package
Package: EnrichGT
Title: EnrichGT - all in one enrichment analysis solution
Version: 1.0.3
Author: Zhiming Ye
Maintainer: Zhiming Ye &lt;garnetcrow@hotmail.com&gt;
Description: Do biological enrichment analysis and parsing and clustering
    enrichment result to insightful results in just ONE package
License: GPL-V3
URL: https://zhimingye.github.io/EnrichGT/
Depends: 
    R (&gt;= 2.10)
Imports: 
    AnnotationDbi,
    BiocManager,
    cli,
    dplyr,
    ellmer,
    fgsea,
    fontawesome,
    forcats,
    ggplot2,
    glue,
    GO.db,
    grDevices,
    gt,
    Matrix,
    methods,
    parallel,
    proxy,
    qvalue,
    RColorBrewer,
    Rcpp,
    reactome.db,
    rlang,
    scales,
    stats,
    stringr,
    text2vec,
    tibble,
    utils,
    xfun
Suggests: 
    devtools,
    readr,
    testthat (&gt;= 3.0.0),
    tidyverse
LinkingTo: 
    Rcpp
Config/testthat/edition: 3
Encoding: UTF-8
LazyData: true
RoxygenNote: 7.3.2</code></pre>
</section>
<section id="convert_annotations_genes-convert-gene-annotations-from-any-keys-to-any-keys" class="level1">
<h1><code>convert_annotations_genes</code>: Convert gene annotations from any keys to any keys</h1>
<section id="description-1" class="level2">
<h2 class="anchored" data-anchor-id="description-1">Description</h2>
<p>Convert gene annotations from any keys to any keys</p>
</section>
<section id="usage" class="level2">
<h2 class="anchored" data-anchor-id="usage">Usage</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">convert_annotations_genes</span>(genes, from_what, to_what, OrgDB)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments" class="level2">
<h2 class="anchored" data-anchor-id="arguments">Arguments</h2>
<ul>
<li><code>genes</code>: gene vector</li>
<li><code>from_what</code>: input type (like “SYMBOL”,“ENTREZID”,“ENSEMBL”,“GENENAME”,…), keys should be supported by AnnotationDbi. Search for the help page of AnnotationDbi for further help.</li>
<li><code>to_what</code>: output type (like “SYMBOL”,“ENTREZID”,“ENSEMBL”,“GENENAME”,…), keys should be supported by AnnotationDbi. Search for the help page of AnnotationDbi for further help. Can be multiple items E.g. <code>c("ENTREZID","ENSEMBL","GENENAME")</code></li>
<li><code>OrgDB</code>: human = org.Hs.eg.db, mouse = org.Mm.eg.db, search BioConductor website for further help</li>
</ul>
</section>
<section id="value" class="level2">
<h2 class="anchored" data-anchor-id="value">Value</h2>
<p>a data.frame</p>
</section>
</section>
<section id="database_from_gmt-parse-gmt-format-gene-set-files" class="level1">
<h1><code>database_from_gmt</code>: Parse GMT format gene set files</h1>
<section id="description-2" class="level2">
<h2 class="anchored" data-anchor-id="description-2">Description</h2>
<p>Reads gene set files in GMT format (e.g., from MSigDB or WikiPathways) and converts them to a data frame suitable for enrichment analysis. Can optionally convert ENTREZ IDs to gene symbols.</p>
</section>
<section id="usage-1" class="level2">
<h2 class="anchored" data-anchor-id="usage-1">Usage</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">database_from_gmt</span>(gmtfile, <span class="at">OrgDB =</span> <span class="cn">NULL</span>, <span class="at">convert_2_symbols =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-1" class="level2">
<h2 class="anchored" data-anchor-id="arguments-1">Arguments</h2>
<ul>
<li><code>gmtfile</code>: Path to GMT format file</li>
<li><code>OrgDB</code>: Annotation database for ID conversion (e.g., org.Hs.eg.db for human). Required if convert_2_symbols=TRUE.</li>
<li><code>convert_2_symbols</code>: Logical indicating whether to convert ENTREZ IDs to gene symbols. Default is TRUE.</li>
</ul>
</section>
<section id="author" class="level2">
<h2 class="anchored" data-anchor-id="author">Author</h2>
<p>Original GMT parser by Guangchuang Yu (https://github.com/YuLab-SMU/gson). Cache system and enhancements by Zhiming Ye.</p>
</section>
<section id="value-1" class="level2">
<h2 class="anchored" data-anchor-id="value-1">Value</h2>
<p>A data frame with columns:</p>
<ul>
<li>term: Gene set name</li>
<li>gene: Gene identifiers (symbols or ENTREZ IDs)</li>
</ul>
<p>If input has 3 columns, includes an additional ID column.</p>
</section>
<section id="examples" class="level2">
<h2 class="anchored" data-anchor-id="examples">Examples</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MSigDB hallmark gene sets</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>gmt_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"h.all.v7.4.symbols.gmt"</span>, <span class="at">package =</span> <span class="st">"EnrichGT"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">database_from_gmt</span>(gmt_file)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Read WikiPathways with ENTREZ to symbol conversion  </span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>gmt_file <span class="ot">&lt;-</span> <span class="st">"wikipathways-20220310-gmt-Homo_sapiens.gmt"</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">database_from_gmt</span>(gmt_file, <span class="at">OrgDB =</span> org.Hs.eg.db)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_compare_groups-2-group-comparison-of-enrichment-results-and-further-clustering-and-visualizing" class="level1">
<h1><code>egt_compare_groups</code>: 2-Group Comparison of enrichment results and further clustering and visualizing</h1>
<section id="description-3" class="level2">
<h2 class="anchored" data-anchor-id="description-3">Description</h2>
<p>See <code>?egt_enrichment_analysis()</code></p>
</section>
<section id="usage-2" class="level2">
<h2 class="anchored" data-anchor-id="usage-2">Usage</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_compare_groups</span>(</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  obj.test,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  obj.ctrl,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">name.test =</span> <span class="cn">NULL</span>,</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">name.ctrl =</span> <span class="cn">NULL</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ClusterNum =</span> <span class="dv">15</span>,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">P.adj =</span> <span class="fl">0.05</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">force =</span> F,</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">nTop =</span> <span class="dv">10</span>,</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-2" class="level2">
<h2 class="anchored" data-anchor-id="arguments-2">Arguments</h2>
<ul>
<li><code>obj.test</code>: the enriched object from tested group. WARNING: <code>obj.test</code> and <code>obj.ctrl</code> should come from same database (e.g.&nbsp;GO Biological Process(GOBP)).</li>
<li><code>obj.ctrl</code>: the enriched object from control group. WARNING: <code>obj.test</code> and <code>obj.ctrl</code> should come from same database (e.g.&nbsp;GO Biological Process(GOBP)).</li>
<li><code>name.test</code>: optional, the name of the testing group. If is <code>NULL</code>, the object name of <code>obj.test</code> will be used.</li>
<li><code>name.ctrl</code>: optional, the name of the control group. If is <code>NULL</code>, the object name of <code>obj.ctrl</code> will be used.</li>
<li><code>ClusterNum</code>: how many cluster will be clustered</li>
<li><code>P.adj</code>: p.adjust cut-off. To avoid slow visualization, you can make stricter p-cut off.</li>
<li><code>force</code>: ignore all auto-self-checks, which is useful</li>
<li><code>nTop</code>: keep n top items according to p-adj in each cluster.</li>
<li><code>method</code>: the agglomeration method to be used. This should be (an unambiguous abbreviation of) one of “ward.D”, “ward.D2”, “single”, “complete”, “average” (= UPGMA), “mcquitty” (= WPGMA), “median” (= WPGMC) or “centroid” (= UPGMC).</li>
<li><code>...</code>: Others options.</li>
</ul>
</section>
<section id="details" class="level2">
<h2 class="anchored" data-anchor-id="details">Details</h2>
<p>Execute <code>obj.test</code> VS <code>obj.ctrl</code> tests, showing pathway overlaps (or differences) and meta-gene modules of test group and control group. Supports ORA and GSEA results (enriched object or data.frame). !WARNING!: <code>obj.test</code> and <code>obj.ctrl</code> should come from same database (e.g.&nbsp;GO Biological Process(GOBP)).</p>
</section>
<section id="author-1" class="level2">
<h2 class="anchored" data-anchor-id="author-1">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-2" class="level2">
<h2 class="anchored" data-anchor-id="value-2">Value</h2>
<p><code>List</code> containing multiple <code>EnrichGT_obj</code> objects. The <code>List</code> contains objects with overlapped enriched terms, unique enrich terms.</p>
</section>
</section>
<section id="egt_enrichment_analysis-perform-over-representation-analysis-ora" class="level1">
<h1><code>egt_enrichment_analysis</code>: Perform Over-Representation Analysis (ORA)</h1>
<section id="description-4" class="level2">
<h2 class="anchored" data-anchor-id="description-4">Description</h2>
<p>ORA compares the proportion of genes in your target list that belong to specific categories (pathways, GO terms etc.) against the expected proportion in a background set. This implementation uses hash tables for efficient gene counting and supports parallel processing for analyzing multiple gene lists.</p>
</section>
<section id="usage-3" class="level2">
<h2 class="anchored" data-anchor-id="usage-3">Usage</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_enrichment_analysis</span>(</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  genes,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  database,</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_adj_methods =</span> <span class="st">"BH"</span>,</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_val_cut_off =</span> <span class="fl">0.5</span>,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">background_genes =</span> <span class="cn">NULL</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">min_geneset_size =</span> <span class="dv">10</span>,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">max_geneset_size =</span> <span class="dv">500</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">multi_cores =</span> <span class="dv">0</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-3" class="level2">
<h2 class="anchored" data-anchor-id="arguments-3">Arguments</h2>
<ul>
<li><code>genes</code>: Input genes, either:</li>
<li>Character vector of gene IDs (e.g., <code>c("TP53","BRCA1")</code>)</li>
<li>Named numeric vector from <code>genes_with_weights()</code> (will split by expression direction)</li>
<li>List of gene vectors for multiple comparisons (e.g., by cell type)</li>
<li><code>database</code>: Gene set database, either:</li>
<li>Built-in database from <code>database_GO_BP()</code>, <code>database_KEGG()</code> etc.</li>
<li>Custom data frame with columns: (ID, Pathway_Name, Genes) or (Pathway_Name, Genes)</li>
<li>GMT file loaded via <code>database_from_gmt()</code></li>
<li><code>p_val_cut_off</code>: Adjusted p-value cutoff (default 0.5)</li>
<li><code>background_genes</code>: Custom background genes (default: all genes in database)</li>
<li><code>min_geneset_size</code>: Minimum genes per set (default 10)</li>
<li><code>max_geneset_size</code>: Maximum genes per set (default 500)</li>
<li><code>multi_cores</code>: (Please don’t use this since it has several known bugs) Number of cores for parallel processing (default 0 = serial)</li>
<li><code>p_adj_method</code>: Multiple testing correction method (default “BH” for Benjamini-Hochberg)</li>
</ul>
</section>
<section id="details-1" class="level2">
<h2 class="anchored" data-anchor-id="details-1">Details</h2>
<p>Identifies enriched biological pathways or gene sets in a gene list using high-performance C++ implementation with parallel processing support.</p>
</section>
<section id="author-2" class="level2">
<h2 class="anchored" data-anchor-id="author-2">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-3" class="level2">
<h2 class="anchored" data-anchor-id="value-3">Value</h2>
<p>A data frame with columns:</p>
<ul>
<li>ID: Gene set identifier</li>
<li>Description: Gene set name</li>
<li>GeneRatio: Enriched genes / input genes</li>
<li>BgRatio: Set genes / background genes</li>
<li>pvalue: Raw p-value</li>
<li>p.adjust: Adjusted p-value</li>
<li>geneID: Enriched genes</li>
<li>Count: Number of enriched genes</li>
</ul>
<p>For weighted input, additional columns show up/down regulated genes.</p>
</section>
<section id="examples-1" class="level2">
<h2 class="anchored" data-anchor-id="examples-1">Examples</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Basic ORA with GO Biological Processes</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TP53"</span>, <span class="st">"BRCA1"</span>, <span class="st">"EGFR"</span>, <span class="st">"CDK2"</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(genes, <span class="fu">database_GO_BP</span>())</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ORA with DEG results (split by direction)</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>deg_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(DEG<span class="sc">$</span>gene, DEG<span class="sc">$</span>log2FC)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(deg_genes, <span class="fu">database_KEGG</span>())</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Multi-group ORA with parallel processing</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>gene_lists <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Macrophages =</span> <span class="fu">c</span>(<span class="st">"CD68"</span>, <span class="st">"CD163"</span>, <span class="st">"CD169"</span>),</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">Fibroblasts =</span> <span class="fu">c</span>(<span class="st">"COL1A1"</span>, <span class="st">"COL1A2"</span>, <span class="st">"ACTA2"</span>)</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_enrichment_analysis</span>(gene_lists, <span class="fu">database_Reactome</span>(), <span class="at">multi_cores=</span><span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_gsea_analysis-perform-gene-set-enrichment-analysis-gsea" class="level1">
<h1><code>egt_gsea_analysis</code>: Perform Gene Set Enrichment Analysis (GSEA)</h1>
<section id="description-5" class="level2">
<h2 class="anchored" data-anchor-id="description-5">Description</h2>
<p>GSEA analyzes whether predefined gene sets show statistically significant enrichment at the top or bottom of a ranked gene list. This implementation uses the fast fgsea algorithm from the fgsea package.</p>
</section>
<section id="usage-4" class="level2">
<h2 class="anchored" data-anchor-id="usage-4">Usage</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_gsea_analysis</span>(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  genes,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  database,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">p_val_cut_off =</span> <span class="fl">0.5</span>,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">min_geneset_size =</span> <span class="dv">10</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">max_geneset_size =</span> <span class="dv">500</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">gseaParam =</span> <span class="dv">1</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-4" class="level2">
<h2 class="anchored" data-anchor-id="arguments-4">Arguments</h2>
<ul>
<li><code>genes</code>: A named numeric vector of ranked genes where:</li>
<li>Names are gene identifiers</li>
<li>Values are ranking metric (e.g., log2 fold change, PCA loading)Must be sorted in descending order (use <code>genes_with_weights()</code> to prepare)</li>
<li><code>database</code>: Gene set database, either:</li>
<li>Built-in database from <code>database_GO_BP()</code>, <code>database_KEGG()</code> etc.</li>
<li>Custom data frame with columns: (ID, Pathway_Name, Genes) or (Pathway_Name, Genes)</li>
<li>GMT file loaded via <code>database_from_gmt()</code></li>
<li><code>p_val_cut_off</code>: Adjusted p-value cutoff (default 0.5)</li>
<li><code>min_geneset_size</code>: Minimum genes per set (default 10)</li>
<li><code>max_geneset_size</code>: Maximum genes per set (default 500)</li>
<li><code>gseaParam</code>: GSEA parameter controlling weight of ranking (default 1)</li>
<li><code>p_adj_method</code>: Multiple testing correction method (default “BH” for Benjamini-Hochberg)</li>
</ul>
</section>
<section id="details-2" class="level2">
<h2 class="anchored" data-anchor-id="details-2">Details</h2>
<p>Identifies enriched biological pathways in a ranked gene list using the fgsea algorithm.</p>
</section>
<section id="author-3" class="level2">
<h2 class="anchored" data-anchor-id="author-3">Author</h2>
<p>Based on fgsea package by Alexey Sergushichev et al.</p>
</section>
<section id="value-4" class="level2">
<h2 class="anchored" data-anchor-id="value-4">Value</h2>
<p>A data frame with columns:</p>
<ul>
<li>ID: Gene set identifier</li>
<li>Description: Gene set name</li>
<li>ES: Enrichment score</li>
<li>NES: Normalized enrichment score</li>
<li>pvalue: Raw p-value</li>
<li>p.adjust: Adjusted p-value</li>
<li>core_enrichment: Leading edge genes</li>
</ul>
</section>
<section id="examples-2" class="level2">
<h2 class="anchored" data-anchor-id="examples-2">Examples</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Using differential expression results</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(DEG<span class="sc">$</span>gene, DEG<span class="sc">$</span>log2FC)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_gsea_analysis</span>(ranked_genes, <span class="fu">database_GO_BP</span>())</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Using PCA loadings</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(<span class="fu">rownames</span>(pca<span class="sc">$</span>rotation), pca<span class="sc">$</span>rotation[,<span class="dv">1</span>])</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_gsea_analysis</span>(ranked_genes, <span class="fu">database_KEGG</span>())</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom gene sets from GMT file</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(genes, weights)</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_gsea_analysis</span>(ranked_genes, <span class="fu">database_from_gmt</span>(<span class="st">"pathways.gmt"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_infer_act-infering-pathway-or-transcript-factors-activity-from-enrichgt-meta-gene-modules" class="level1">
<h1><code>egt_infer_act</code>: Infering Pathway or Transcript Factors activity from EnrichGT meta-gene modules</h1>
<section id="description-6" class="level2">
<h2 class="anchored" data-anchor-id="description-6">Description</h2>
<p>Only supports gene symbols. <a href="https://saezlab.github.io/progeny/">PROGENy</a> is a comprehensive resource containing a curated collection of pathways and their target genes, with weights for each interaction. <a href="https://github.com/saezlab/CollecTRI">CollecTRI</a> is a comprehensive resource containing a curated collection of TFs and their transcriptional targets compiled from 12 different resources. This collection provides an increased coverage of transcription factors and a superior performance in identifying perturbed TFs compared to our previous. If when doing re-enrichment, you select a high number of clusters, that may cause low gene number in each meta-gene module, and then can’t be infered sucessfully. So if result is empty, please increase the number of re-clustering when doing it.</p>
</section>
<section id="usage-5" class="level2">
<h2 class="anchored" data-anchor-id="usage-5">Usage</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_infer_act</span>(x, <span class="at">DB =</span> <span class="st">"collectri"</span>, <span class="at">species =</span> <span class="st">"human"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-5" class="level2">
<h2 class="anchored" data-anchor-id="arguments-5">Arguments</h2>
<ul>
<li><code>x</code>: an EnrichGT_obj object.</li>
<li><code>DB</code>: can be “progeny” (the Pathway activity database), or “collectri” (TF activity database)</li>
<li><code>species</code>: can be “human” or “mouse”</li>
</ul>
</section>
<section id="author-4" class="level2">
<h2 class="anchored" data-anchor-id="author-4">Author</h2>
<p>Zhiming Ye, Saez-Rodriguez Lab (The decoupleR package, https://saezlab.github.io/decoupleR/)</p>
</section>
<section id="value-5" class="level2">
<h2 class="anchored" data-anchor-id="value-5">Value</h2>
<p>an ORA result list</p>
</section>
</section>
<section id="egt_llm_summary-summarize-enrichgt-results-using-llm" class="level1">
<h1><code>egt_llm_summary</code>: Summarize EnrichGT results using LLM</h1>
<section id="description-7" class="level2">
<h2 class="anchored" data-anchor-id="description-7">Description</h2>
<p>This function uses a Large Language Model (LLM) to generate summaries for pathway clusters and gene modules in an EnrichGT_obj object.</p>
</section>
<section id="usage-6" class="level2">
<h2 class="anchored" data-anchor-id="usage-6">Usage</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_llm_summary</span>(x, chat, <span class="at">lang =</span> <span class="st">"English"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-6" class="level2">
<h2 class="anchored" data-anchor-id="arguments-6">Arguments</h2>
<ul>
<li><code>x</code>: An EnrichGT_obj object created by <code>[egt_recluster_analysis](egt_recluster_analysis)</code>.</li>
<li><code>chat</code>: An LLM chat object created by the <code>ellmer</code> package.</li>
<li><code>lang</code>: Language pass to LLM. Can be <code>English</code> or <code>Chinese</code>.</li>
</ul>
</section>
<section id="note" class="level2">
<h2 class="anchored" data-anchor-id="note">Note</h2>
<p>It is recommended not to add system prompts when creating the chat object. The function provides its own carefully crafted prompts for biological analysis.</p>
</section>
<section id="references" class="level2">
<h2 class="anchored" data-anchor-id="references">References</h2>
<p>For more information about creating chat objects, see the <a href="https://ellmer.tidyverse.org/index.html">ellmer package documentation</a>.</p>
</section>
<section id="seealso" class="level2">
<h2 class="anchored" data-anchor-id="seealso">Seealso</h2>
<p><code>[egt_recluster_analysis](egt_recluster_analysis)</code> to create the input object.</p>
</section>
<section id="value-6" class="level2">
<h2 class="anchored" data-anchor-id="value-6">Value</h2>
<p>Returns the input EnrichGT_obj object with added LLM annotations in the <code>LLM_Annotation</code> slot. The annotations include:</p>
<ul>
<li><code>pathways</code>: Summaries of pathway clusters</li>
<li><code>genes_and_title</code>: Summaries of gene modules and their titles</li>
</ul>
</section>
<section id="examples-3" class="level2">
<h2 class="anchored" data-anchor-id="examples-3">Examples</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create LLM chat object</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>chat <span class="ot">&lt;-</span> <span class="fu">chat_deepseek</span>(<span class="at">api_key =</span> YOUR_API_KEY, <span class="at">model =</span> <span class="st">"deepseek-chat"</span>, <span class="at">system_prompt =</span> <span class="st">""</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis and get EnrichGT_obj</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>re_enrichment_results <span class="ot">&lt;-</span> <span class="fu">egt_recluster_analysis</span>(...)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get LLM summaries</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>re_enrichment_results <span class="ot">&lt;-</span> <span class="fu">egt_llm_summary</span>(re_enrichment_results, chat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="egt_plot_gsea-generate-gsea-enrichment-plots" class="level1">
<h1><code>egt_plot_gsea</code>: Generate GSEA Enrichment Plots</h1>
<section id="description-8" class="level2">
<h2 class="anchored" data-anchor-id="description-8">Description</h2>
<p>This function creates graphical representations of Gene Set Enrichment Analysis (GSEA) results, including either a multi-panel GSEA table plot for multiple pathways or a single pathway enrichment plot. The visualization leverages the <code>fgsea</code> package’s plotting functions.</p>
</section>
<section id="usage-7" class="level2">
<h2 class="anchored" data-anchor-id="usage-7">Usage</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_gsea</span>(resGSEA<span class="sc">$</span>Description[<span class="dv">1</span>],<span class="at">genes =</span> <span class="fu">genes_with_weights</span>(<span class="at">genes =</span> DEGexample<span class="sc">$</span>...<span class="dv">1</span>, <span class="at">weights =</span> DEGexample<span class="sc">$</span>log2FoldChange),<span class="at">database =</span> <span class="fu">database_GO_BP</span>(org.Hs.eg.db))</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_gsea</span>(resGSEA[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,],<span class="at">genes =</span> <span class="fu">genes_with_weights</span>(<span class="at">genes =</span> DEGexample<span class="sc">$</span>...<span class="dv">1</span>, <span class="at">weights =</span> DEGexample<span class="sc">$</span>log2FoldChange),<span class="at">database =</span> <span class="fu">database_GO_BP</span>(org.Hs.eg.db))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-7" class="level2">
<h2 class="anchored" data-anchor-id="arguments-7">Arguments</h2>
<ul>
<li><code>x</code>: A GSEA result object. Can be either:</li>
<li>A data frame containing GSEA results (requires columns: pvalue, p.adjust, Description)</li>
<li>A character string specifying a single pathway name</li>
<li><code>genes</code>: A named numeric vector from <code>genes_with_weights()</code>. These should match the gene identifiers used in the GSEA analysis.</li>
<li><code>database</code>: A database <code>data.frame</code>, You can obtain it from <code>database_xxx()</code> functions. This should correspond to the database used in the original GSEA analysis.</li>
</ul>
</section>
<section id="author-5" class="level2">
<h2 class="anchored" data-anchor-id="author-5">Author</h2>
<p>Zhiming Ye, warpped from <code>fgsea</code></p>
</section>
<section id="value-7" class="level2">
<h2 class="anchored" data-anchor-id="value-7">Value</h2>
<p>A ggplot object:</p>
<ul>
<li>When <code>x</code> is a data frame: Returns a multi-panel plot showing normalized enrichment scores (NES), p-values, and leading edge plots for top pathways</li>
<li>When <code>x</code> is a pathway name: Returns an enrichment plot showing the running enrichment score for the specified pathway</li>
</ul>
</section>
</section>
<section id="egt_plot_results-visualize-enrichment-results-using-simple-plot" class="level1">
<h1><code>egt_plot_results</code>: Visualize enrichment results using simple plot</h1>
<section id="description-9" class="level2">
<h2 class="anchored" data-anchor-id="description-9">Description</h2>
<p>This plot is the most widely like <code>enrichplot::dotplot()</code>used method to visualize enriched terms. It shows the enrichment scores (e.g.&nbsp;p values) and gene ratio or NES as dot size and color / or bar height. Users can specify the number of terms using <code>ntop</code> or selected terms to color via the <code>low.col</code> and <code>hi.col</code>.</p>
</section>
<section id="usage-8" class="level2">
<h2 class="anchored" data-anchor-id="usage-8">Usage</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_plot_results</span>(</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ntop =</span> <span class="cn">NULL</span>,</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">showIDs =</span> F,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">max_len_descript =</span> <span class="dv">40</span>,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">keepAll =</span> F,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  ...,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">P.adj =</span> <span class="cn">NULL</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-8" class="level2">
<h2 class="anchored" data-anchor-id="arguments-8">Arguments</h2>
<ul>
<li><code>x</code>: a data frame form enriched result like <code>egt_enrichment_analysis()</code> or <code>egt_gsea_analysis()</code>, or an re-clustered <code>EnrichGT</code> object</li>
<li><code>ntop</code>: Show top N in each cluster. In default, for origin enriched result, showing top 15; for re-clustered object, showing top 5 in each cluster.</li>
<li><code>showIDs</code>: bool, show pathway IDs or not. Default is FALSE</li>
<li><code>max_len_descript</code>: the label format length, default as 40.</li>
<li><code>...</code>: Other param</li>
<li><code>P.adj</code>: If pass an origin data.frame from original enriched result, you can specify the P-adjust value cut off. If is null, default is 0.05. When passing <code>EnrichGT_obj</code>, this filter is previously done by <code>egt_recluster_analysis</code>.</li>
<li><code>low.col</code>: the color for the lowest</li>
<li><code>hi.col</code>: the color for the highest</li>
</ul>
</section>
<section id="author-6" class="level2">
<h2 class="anchored" data-anchor-id="author-6">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-8" class="level2">
<h2 class="anchored" data-anchor-id="value-8">Value</h2>
<p>a ggplot2 object</p>
</section>
</section>
<section id="egt_recluster_analysis-cluster-and-re-enrichment-enrichment-results" class="level1">
<h1><code>egt_recluster_analysis</code>: Cluster and re-enrichment enrichment results</h1>
<section id="description-10" class="level2">
<h2 class="anchored" data-anchor-id="description-10">Description</h2>
<p>Performs hierarchical clustering on enrichment results (ORA or GSEA) based on gene-term associations to reduce redundancy and improve biological interpretation. The function helps identify coherent groups of related terms while preserving important but less significant findings.</p>
</section>
<section id="usage-9" class="level2">
<h2 class="anchored" data-anchor-id="usage-9">Usage</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">egt_recluster_analysis</span>(</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  x,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ClusterNum =</span> <span class="dv">10</span>,</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">P.adj =</span> <span class="fl">0.05</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">force =</span> F,</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">nTop =</span> <span class="dv">10</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-9" class="level2">
<h2 class="anchored" data-anchor-id="arguments-9">Arguments</h2>
<ul>
<li><code>x</code>: Enrichment result from <code>EnrichGT</code> or <code>clusterProfiler</code>. For multi-database results, provide a <code>list</code>.</li>
<li><code>ClusterNum</code>: Number of clusters to create (default: 10).</li>
<li><code>P.adj</code>: Adjusted p-value cutoff (default: 0.05). Stricter values improve performance.</li>
<li><code>force</code>: Logical to bypass validation checks (default: FALSE).</li>
<li><code>nTop</code>: Number of top terms to keep per cluster by p-value (default: 10).</li>
<li><code>method</code>: Hierarchical clustering method (default: “ward.D2”). One of: “ward.D”, “ward.D2”, “single”, “complete”, “average” (UPGMA), “mcquitty” (WPGMA), “median” (WPGMC), or “centroid” (UPGMC).</li>
<li><code>...</code>: Additional arguments passed to clustering functions.</li>
</ul>
</section>
<section id="details-3" class="level2">
<h2 class="anchored" data-anchor-id="details-3">Details</h2>
<p>Input requirements by analysis type: ORA results: Required columns: “ID”, “Description”, “GeneRatio”, “pvalue”, “p.adjust”, “geneID”, “Count” GSEA results:<br>
Required columns: “ID”, “Description”, “NES”, “pvalue”, “p.adjust”, “core_enrichment” compareClusterResult: Either the compareClusterResult object or a data frame with: - All ORA columns listed above - Additional “Cluster” column Multi-database: Provide as a named list of the above result types</p>
</section>
<section id="author-7" class="level2">
<h2 class="anchored" data-anchor-id="author-7">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-9" class="level2">
<h2 class="anchored" data-anchor-id="value-9">Value</h2>
<p>An <code>EnrichGT_obj</code> containing:</p>
<ul>
<li>enriched_result: Filtered results data frame</li>
<li>gt_object: Formatted <code>gt</code> table object</li>
<li>gene_modules: List of gene modules per cluster</li>
<li>pathway_clusters: Pathway names by cluster</li>
<li>clustering_tree: <code>hclust</code> object for visualization</li>
<li>raw_enriched_result: Unfiltered results table</li>
</ul>
</section>
<section id="examples-4" class="level2">
<h2 class="anchored" data-anchor-id="examples-4">Examples</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ORA example</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">egt_recluster_analysis</span>(ora_result, <span class="at">ClusterNum=</span><span class="dv">8</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(res<span class="sc">@</span>clustering_tree)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># GSEA example </span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>gsea_res <span class="ot">&lt;-</span> <span class="fu">egt_recluster_analysis</span>(gsea_result, <span class="at">method=</span><span class="st">"average"</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">gtsave</span>(gsea_res<span class="sc">@</span>gt_object, <span class="st">"results.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="genes_with_weights-create-a-ranked-gene-list-for-gsea-analysis" class="level1">
<h1><code>genes_with_weights</code>: Create a ranked gene list for GSEA analysis</h1>
<section id="description-11" class="level2">
<h2 class="anchored" data-anchor-id="description-11">Description</h2>
<p>Takes gene identifiers and corresponding weights (like log2 fold changes) and returns a ranked vector suitable for Gene Set Enrichment Analysis (GSEA).</p>
</section>
<section id="usage-10" class="level2">
<h2 class="anchored" data-anchor-id="usage-10">Usage</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">genes_with_weights</span>(genes, weights)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-10" class="level2">
<h2 class="anchored" data-anchor-id="arguments-10">Arguments</h2>
<ul>
<li><code>genes</code>: Character vector of gene identifiers (e.g., gene symbols or ENTREZ IDs)</li>
<li><code>weights</code>: Numeric vector of weights for each gene (typically log2 fold changes)</li>
</ul>
</section>
<section id="author-8" class="level2">
<h2 class="anchored" data-anchor-id="author-8">Author</h2>
<p>Zhiming Ye</p>
</section>
<section id="value-10" class="level2">
<h2 class="anchored" data-anchor-id="value-10">Value</h2>
<p>A named numeric vector sorted in descending order by weight, where: - Names are gene identifiers<br>
- Values are the corresponding weights</p>
</section>
<section id="examples-5" class="level2">
<h2 class="anchored" data-anchor-id="examples-5">Examples</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example using differential expression results  </span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TP53"</span>, <span class="st">"BRCA1"</span>, <span class="st">"EGFR"</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>log2fc <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="sc">-</span><span class="fl">2.1</span>, <span class="fl">0.8</span>) </span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>ranked_genes <span class="ot">&lt;-</span> <span class="fu">genes_with_weights</span>(genes, log2fc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="database_...-get-database-for-enrichment-analysis" class="level1">
<h1><code>database_...</code>: Get database for enrichment analysis</h1>
<section id="description-12" class="level2">
<h2 class="anchored" data-anchor-id="description-12">Description</h2>
<p>Get Gene Ontology (GO), Reactome, and other term-to-gene database, for enrichment analysis</p>
</section>
<section id="usage-11" class="level2">
<h2 class="anchored" data-anchor-id="usage-11">Usage</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_BP</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_CC</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_MF</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="fu">database_GO_ALL</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="fu">database_Reactome</span>(<span class="at">OrgDB =</span> org.Hs.eg.db)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="fu">database_progeny_human</span>()</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="fu">database_progeny_mouse</span>()</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="fu">database_CollecTRI_human</span>()</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="fu">database_CollecTRI_mouse</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-11" class="level2">
<h2 class="anchored" data-anchor-id="arguments-11">Arguments</h2>
<ul>
<li><code>OrgDB</code>: The AnnotationDbi database to fetch pathway data and convert gene IDs to gene symbols. For human it would be <code>org.Hs.eg.db</code>, for mouse it would be <code>org.Mm.eg.db</code>. In AnnotationDbi there are many species, please search <code>AnnotationDbi</code> for other species annotation database. GO and Reactome should add this, progeny and collectri do not.</li>
</ul>
</section>
<section id="author-9" class="level2">
<h2 class="anchored" data-anchor-id="author-9">Author</h2>
<p>Zhiming Ye. Part of functions were inspired by <code>clusterProfiler</code> but with brand new implement.</p>
</section>
<section id="value-11" class="level2">
<h2 class="anchored" data-anchor-id="value-11">Value</h2>
<p>a data.frame with ID, terms and genes</p>
</section>
</section>
<section id="database_kegg-get-kegg-database-from-kegg-website" class="level1">
<h1><code>database_KEGG</code>: Get KEGG database from KEGG website</h1>
<section id="description-13" class="level2">
<h2 class="anchored" data-anchor-id="description-13">Description</h2>
<p>KEGG is a commercialized database. So EnrichGT can’t pre-cache them locally. You can use this function to fetch KEGG database pathways and modules.</p>
</section>
<section id="usage-12" class="level2">
<h2 class="anchored" data-anchor-id="usage-12">Usage</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">database_KEGG</span>(<span class="at">kegg_organism=</span><span class="st">"hsa"</span>,<span class="at">OrgDB =</span> org.Hs.eg.db,<span class="at">kegg_modules=</span>F,<span class="at">local_cache=</span>F)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">database_KEGG_show_organism</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-12" class="level2">
<h2 class="anchored" data-anchor-id="arguments-12">Arguments</h2>
<ul>
<li><code>kegg_organism</code>: Determine which species data from KEGG will be fetch. For human, it would be <code>hsa</code>(in default); For mouse, it would be <code>mmu</code>. If you wants other species, see <code>database_kegg_show_organism()</code> for details.</li>
<li><code>OrgDB</code>: The AnnotationDbi database to convert KEGG gene ID to gene symbols. For human it would be <code>org.Hs.eg.db</code>, for mouse it would be <code>org.Mm.eg.db</code>. In AnnotationDbi there are many species, please search <code>AnnotationDbi</code> for other species annotation database.</li>
<li><code>kegg_modules</code>: If TRUE, returns KEGG module; If FALSE returns KEGG pathways. In default, this is setted to FALSE to get mouse commonly used KEGG pathways.</li>
<li><code>local_cache</code>: cache a copy in local working folder. It will be saved as a <code>.enrichgt_cache</code> file in working dictionary. The <code>.enrichgt_cache</code> is just a <code>.rds</code> file, feel free to read it using <code>readRDS()</code>.</li>
</ul>
</section>
</section>
<section id="delete--filter-enrichment-results-by-description-pattern" class="level1">
<h1><code>%-delete-&gt;%</code>: Filter Enrichment Results by Description Pattern</h1>
<section id="description-14" class="level2">
<h2 class="anchored" data-anchor-id="description-14">Description</h2>
<p>Infix operator to filter enrichment results by matching against Description field. For EnrichGT_obj objects, re-runs clustering analysis after filtering.</p>
</section>
<section id="usage-13" class="level2">
<h2 class="anchored" data-anchor-id="usage-13">Usage</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>x <span class="sc">%-delete-&gt;%</span> y</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="arguments-13" class="level2">
<h2 class="anchored" data-anchor-id="arguments-13">Arguments</h2>
<ul>
<li><code>x</code>: Either an EnrichGT_obj object or data.frame containing enrichment results</li>
<li><code>y</code>: Regular expression pattern to match against Description field</li>
</ul>
</section>
<section id="details-4" class="level2">
<h2 class="anchored" data-anchor-id="details-4">Details</h2>
<p>This operator helps refine enrichment results by removing terms matching the given pattern from the Description field. When applied to EnrichGT_obj, it preserves all original parameters and re-runs the clustering analysis on the filtered results.</p>
</section>
<section id="value-12" class="level2">
<h2 class="anchored" data-anchor-id="value-12">Value</h2>
<p>For EnrichGT_obj input: A new EnrichGT_obj with filtered and re-clustered results. For data.frame input: A filtered data.frame.</p>
</section>
<section id="examples-6" class="level2">
<h2 class="anchored" data-anchor-id="examples-6">Examples</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out "ribosome" related terms</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>filtered_results <span class="ot">&lt;-</span> reenrichment_obj <span class="sc">%-delete-&gt;%</span> <span class="st">"ribosome"</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data.frame directly</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>filtered_df <span class="ot">&lt;-</span> df <span class="sc">%-delete-&gt;%</span> <span class="st">"metabolism"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>


</section>
</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Developed by Zhiming Ye</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ZhimingYe/EnrichGT_website/blob/main/pkgfuns.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/ZhimingYe/EnrichGT_website/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Corresponding to EnrichGT Version 0.8.6</p>
</div>
  </div>
</footer>




</body></html>